docs/mgmt/readme.md

```diff
@@ -0,0 +1,660 @@
+```
+
+If all the above efforts are not met, you need to use the `list-exception` configuration to temporarily disable this check for this collection class:
+```yaml
+list-exception:
+- /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/availabilitySets/{availabilitySetName}
+```
+
+### Scope resources
+
+Some resources are scope resources, which means that these resources can be created under different scopes, like subscriptions, resource groups, management groups, etc. In the swagger, we currently do not have an extension which assigns which type of resources can be the scope of this resource, therefore we add a configuration in our generator `request-path-to-scope-resource-types` for this new information.
+
+By default, the generator will recognize the first parameter in request path as a scope parameter if the parameter has `x-ms-skip-url-encoding: true` on it, and the generator will generate the resource assuming the scope can be anything. For instance, if we have a request path like this:
+```
+/{scope}/providers/Microsoft.Resources/deployments/{deploymentName}
+```
+and we should have a resource generated like this:
+```csharp
+public partial class DeploymentResource : ArmResource
+{
+    /* ... */
+}
+
+public partial class DeploymentCollection : ArmCollection, IEnumerable<DeploymentResource>, IAsyncEnumerable<DeploymentResource>
+{
+    /* ... */
+}
+
+public partial class DeploymentData
+{
+    /* ... */
+}
+```
+and since it is a scope resource without any configuration, its parent will be marked as anything, and we should have the following extension method:
+```csharp
+public static partial class ResourcesExtension
+{
+    public static DeploymentCollection GetDeployments(this ArmResource armResource)
+    {
+        /* ... */
+    }
+}
+```
+To assign specific resource types to this scope, you can use the following configuration:
+
+```yaml
+request-path-to-scope-resource-types:
+  /{scope}/providers/Microsoft.Resources/deployments/{deploymentName}:
+    - subscriptions
+    - resourceGroups
+    - managementGroups
+```
+
+After we applied this configuration, we will have the following changes:
+```diff
+public static partial class ResourcesExtension
+{
+-   public static DeploymentCollection GetDeployments(this ArmResource armResource)
+-   {
+-       /* ... */
+-   }
++   public static DeploymentCollection GetDeployments(this SubscriptionResource subscription)
++   {
++       /* ... */
++   }
++   public static DeploymentCollection GetDeployments(this ResourceGroupResource resourceGroup)
++   {
++       /* ... */
++   }
++   public static DeploymentCollection GetDeployments(this ManagementGroupResource managementGroup)
++   {
++       /* ... */
++   }
+}
+```
+
+### Management debug options
+
+A debug configuration is also introduced to show some extra information in the generated code. Usage:
+
+```
+mgmt-debug:
+  suppress-list-exception: true
+```
+
```

Comment:
how about
- rename-rules
- change the format
- property rename 
- model rename
- enum rename

Reply:
I will add `rename-rules`.

Others are directives brought by the autorest core functionality, I do not think we should introduce that in this documentation. We should document that elsewhere